src/mpyl/steps/collection.py

```diff
@@ -0,0 +1,33 @@
+"""A collection of all available step executors."""
+import importlib
+import pkgutil
+from logging import Logger
+from typing import Optional
+
+from . import Step, IPluginRegistry
+from ..project import Stage
+
+
+class StepsCollection:
+    _step_executors: set[Step]
+
+    def __init__(self, logger: Logger, base_path: Optional[str] = None) -> None:
+        self._step_executors = set()
+        self._base_path = base_path
```

Comment:
Just curious: is there a reason `_step_executors` gets initiated above but `_base_path` not?